### PR TITLE
Create API gateway for the SAS token dispensing endpoint

### PR DESCRIPTION
**Pull request by lgonczar**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RPE-545

### Change description ###

Create a subscription-based API gateway for the SAS token dispensing endpoint

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: Jenkinsfile_CNP [hunk 1]
```diff
@@ -8,10 +8,28 @@ def component = "bulk-scan-processor"
 
 def channel = '#rpe-build-notices'
 
+List<LinkedHashMap<String, Object>> apiTestSecrets = [
```
Comment (rkondratowicz) [inline Jenkinsfile_CNP]: comment would be appreciated

File: README.md [hunk 2]
```diff
@@ -67,6 +67,42 @@ Smoke tests expect an address of deployed application to be passed in `TEST_URL`
...
+Bulk Scan Processor uses an (Azure API Management) API to protect its SAS token dispensing endpoint.
+The API allows only HTTPS requests with a valid subscription key.
+
+Jenkins (pipeline) runs the API gateway tests by executing `apiGateway` gradle task. This only happens if
+there's a call to `enableApiGatewayTest()` in your Jenkinsfile_CNP/Jenkinsfile_parameterized.
+
+Your API tests rely on the following environment variables:
+- `TEST_CLIENT_SUBSCRIPTION_KEY` - subscription key that allows access to the API. Must be set manually in Azure Key Vault.
+- `API_GATEWAY_URL` - The URL of the API (gateway) that is the target of tests.
+This is provided by Jenkins based on the `api_gateway_url` output variable, defined in Terraform code.
```
Comment (rkondratowicz) [inline README.md]: this is all done already, to we need to put it in readme?

File: README.md [hunk 3]
```diff
@@ -67,6 +67,42 @@ Smoke tests expect an address of deployed application to be passed in `TEST_URL`
...
+
+**Store subscription key in Azure Key Vault so that API tests can use it**
+
+```
+az keyvault secret set --vault-name rpe-bsp-{environment} --name test-client-subscription-key --value {the subscription key}
+```
+
+When running the pipeline, Jenkins will read this secret and convert it to `TEST_CLIENT_SUBSCRIPTION_KEY`
+environment variable, available to tests.
+
```
Comment (rkondratowicz) [inline README.md]: Perhaps a link to confluence page would be more suitable?
Otherwise this section will be repeated in all the repos that sue gatewats.

Comment (lgonczar) [reply README.md]: Not sure - this solution is specific to this project.

File: README.md (hunk 2 above)
Comment (lgonczar) [reply README.md]: Not sure. When the subscription key expires or the service is recreated, the subscription key will have to be stored again. What do you think?

**Comment by lgonczar** [comment]
Closing - scope changed

**Pull request closed without merging**